test/trento/domain/sap_system/sap_system_test.exs

```diff
@@ -26,7 +26,93 @@ defmodule Trento.SapSystemTest do
   alias Trento.Domain.SapSystem
 
   describe "SAP System registration" do
-    test "should create an incomplete SAP system aggregate and register a database instance" do
+    test "should fail when a sap system not exists and the database instance has Secondary role" do
```

Comment:
I would add a new test case to see that Secondary nodes are registered once the primary is there.
I think we miss this part, and it would be good to check

Reply:
we already have this test: https://github.com/trento-project/web/blob/9f8e805d08433d0b53a584f7ba3802467840f709/test/trento/domain/sap_system/sap_system_test.exs#L190 

I wouldn't add an extra test since it will be redundant

Reply:
Alright! It is not explicit, but good enough. I buy your point hehe